arm-storageimportexport/2016-11-01/swagger/storageimportexport.json

```diff
@@ -0,0 +1,1209 @@
+    "version": "2016-11-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.windows-ppe.net/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.ImportExport/locations": {
+      "get": {
+        "tags": [
+          "StorageImportExport"
+        ],
+        "operationId": "ListLocations",
+        "description": "Returns a list of locations to which you can ship the disks associated with an import or export job. A location is a Microsoft data center region.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/Accept-Language"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/LocationsListResult"
+            }
+          },
+          "default": {
+            "description": "An error occurs.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.ImportExport/locations/{locationName}": {
+      "get": {
+        "tags": [
+          "StorageImportExport"
+        ],
+        "operationId": "GetLocation",
+        "description": "Gets a location to which you can ship the disks associated with an import or export job. A location is an Azure region.",
+        "parameters": [
+          {
+            "name": "locationName",
+            "in": "path",
+            "description": "The name of the location. For example, “West US” or “westus”.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/Accept-Language"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/Location"
+            }
+          },
+          "default": {
+            "description": "An error occurs.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ImportExport/jobs": {
+      "get": {
+        "tags": [
+          "StorageImportExport"
+        ],
+        "operationId": "Jobs_ListBySubscription",
```

Comment:
The convention that we have chosen is to use the name List for listing resources in a subscription and listByResourceGroup for listing resources in a resource group. Please take a look at the recommended patterns over [here](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md#swagger-recommended-patterns).

This in particular

Consistency in the naming of list methods
Operations that provide resources in a resource group should be named as "OperationGroup_ListByResourceGroup"
Operations that provide resources in a subscription should be named as "OperationGroup_List"
If the ListByResourceGroup is overloaded "i.e. if the resourcegroup is null then the operation lists all resources in the subscription" then, such operations should not be allowed.
If the list operation is on a child resource type then please follow the convention of Noun_ListBy{Parent}. For example "Databases" is a child resource type of "SqlServer". So for listing all the databases the operationId should be Databases_ListBySqlServer. This will ensure consistency in the naming pattern across all the RPs in Azure and it will be easy for customers onbaording to Azure (reduces their learning curve).